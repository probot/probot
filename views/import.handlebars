<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Import {{#if pkg.name }}{{ pkg.name }}{{else}}Your App{{/if}} | built with Probot</title>
    <link rel="icon" href="/probot/static/probot-head.png">
    <link rel="stylesheet" href="/probot/static/primer.css">
</head>

<body class="bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full py-6">
        <a href="/probot"><img src="/probot/static/robot.svg" alt="Probot Logo" width="100" class="mb-6"></a>
        <div class="box-shadow rounded-2 border p-6 bg-white">
            <h2>Use existing Github App</h2>
            <br>

            <h3>Step 1:</h3>
            <p class="d-block mt-2">
                Replace your app's Webhook URL with <br>
                <b>{{ WEBHOOK_PROXY_URL }}</b>
            </p>
            <a class="d-block mt-2" href="{{ GH_HOST }}/settings/apps" target="__blank" rel="noreferrer">
                You can do it here
            </a>

            <br>
            <h3>Step 2:</h3>
            <p class="mt-2">Fill out this form</p>
            <form onsubmit="return onSubmit(event) || false">
                <label class="d-block mt-2" for="appId">App Id</label>
                <input class="form-control width-full" type="text" required="true" id="appId" name="appId"><br>

                <label class="d-block mt-3" for="whs">Webhook secret (required!)</label>
                <input class="form-control width-full" type="password" required="true" id="whs" name="whs"><br>

                <label class="d-block mt-3" for="pem">Private Key</label>
                <input class="form-control width-full m-2" type="file" accept=".pem" required="true" id="pem"
                    name="pem">
                <br>

                <button class="btn btn-outline m-2" type="submit">Submit</button>
            </form>
        </div>

        <div class="mt-4">
            <h4 class="alt-h4 text-gray-light">Need help?</h4>
            <div class="d-flex flex-justify-center mt-2">
                <a href="https://probot.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
                <a href="https://probot-slackin.herokuapp.com/" class="btn btn-outline">Chat on Slack</a>
            </div>
        </div>
    </div>
    <script>
        function onSubmit(e) {
            e.preventDefault();

            const idEl = document.getElementById('appId');
            const appId = idEl.value;


            const secretEl = document.getElementById('whs');
            const webhook_secret = secretEl.value;

            const fileEl = document.getElementById('pem');
            const file = fileEl.files[0];

            file.text().then((text) => fetch('', {
                method: 'POST',
                headers: { 'content-type': 'application/json' },
                body: JSON.stringify({ appId, pem: text, webhook_secret })
            })).then((r) => {
                if (r.ok) {
                    location.replace('/probot/success');
                }
            }).catch((e) => alert(e));
            return false;
        }
    </script>
</body>

</html>