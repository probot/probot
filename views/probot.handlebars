<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>{{#if name }}{{ name }}{{else}}Your App{{/if}} | built with Probot</title>
    <link rel="icon" href="/probot/static/probot-head.png">
    <link rel="stylesheet" href="/probot/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/probot/static/robot.svg" alt="Probot Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to {{#if name }}{{ name }}{{else}}your Probot App{{/if}}
          {{#if version}}
            <span class="Label Label--outline v-align-middle ml-2 text-gray-light">v{{ version }}</span>
          {{/if}}
        </h1>

        {{#if description }}
          <p>{{ description }}</p>
        {{else}}
          <p>This bot was built using <a href="https://github.com/probot/probot">Probot</a>, a framework for building GitHub Apps.</p>
        {{/if}}
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://probot.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://probot-slackin.herokuapp.com/" class="btn btn-outline">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>
