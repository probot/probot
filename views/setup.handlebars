<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Setup {{#if pkg.name }}{{ pkg.name }}{{else}}Your App{{/if}} | built with Probot</title>
    <link rel="icon" href="/probot/static/probot-head.png">
    <link rel="stylesheet" href="/probot/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/probot/static/robot.svg" alt="Probot Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to {{#if pkg.name }}{{ pkg.name }}{{else}}your Probot App{{/if}}
          {{#if version}}
            <span class="Label Label--outline v-align-middle ml-2 text-gray-light">v{{ pkg.version }}</span>
          {{/if}}
        </h1>

        {{#if pkg.description }}
          <p>{{ pkg.description }}</p>
        {{else}}
          <p>This app was built using <a href="https://github.com/probot/probot">Probot</a>, a framework for building GitHub Apps.</p>
        {{/if}}

        <div class="text-left mt-6">
          <h2 class="alt-h3 mb-2">Getting Started</h2>

          <p>To start building a GitHub App, you'll need to register a new app on GitHub.</p>
          <br>

          <form action="{{ createAppUrl }}" method="post" target="_blank" class="d-flex flex-items-center">
            <button class="btn btn-outline" name="manifest" id="manifest" value='{{ manifest }}' >Register GitHub App</button>
            <a href="/probot/import" class="ml-2">or use an existing Github App</a>
          </form>
        </div>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://probot.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://probot-slackin.herokuapp.com/" class="btn btn-outline">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>
