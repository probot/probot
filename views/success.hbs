<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Setup {{#if pkg.name }}{{ pkg.name }}{{else}}Your App{{/if}} | built with Probot</title>
    <link rel="icon" href="/probot/static/probot-head.png">
    <link rel="stylesheet" href="/probot/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/probot/static/robot.svg" alt="Probot Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <div class="text-center">
          <h1 class="alt-h3 mb-2">Congrats! You have successfully installed your app!
            <br>
            Checkout <a href="https://probot.github.io/docs/webhooks/">Receiving webhooks</a> and <a href="https://probot.github.io/docs/github-api/">Interacting with GitHub</a> to learn more!</h1>
        </div>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://probot.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://probot-slackin.herokuapp.com/" class="btn btn-outline">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>
