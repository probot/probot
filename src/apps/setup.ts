import bodyParser from "body-parser";
import { exec } from "child_process";
import { Request, Response } from "express";
import updateDotenv from "update-dotenv";

import { Probot } from "../probot";
import { ManifestCreation } from "../manifest-creation";
import { getLoggingMiddleware } from "../server/logging-middleware";
import { ApplicationFunctionOptions } from "../types";

export const setupAppFactory = (
  host: string | undefined,
  port: number | undefined
) =>
  async function setupApp(
    app: Probot,
    { getRouter }: ApplicationFunctionOptions
  ) {
    const setup: ManifestCreation = new ManifestCreation();

    // If not on Glitch or Production, create a smee URL
    if (
      process.env.NODE_ENV !== "production" &&
      !(process.env.PROJECT_DOMAIN || process.env.WEBHOOK_PROXY_URL)
    ) {
      await setup.createWebhookChannel();
    }

    if (!getRouter) {
      throw new Error("getRouter is required to use the setup app");
    }

    const route = getRouter();

    route.use(getLoggingMiddleware(app.log));

    printWelcomeMessage(app, host, port);

    route.get("/probot", async (req, res) => {
      const baseUrl = getBaseUrl(req);
      const pkg = setup.pkg;
      const manifest = setup.getManifest(pkg, baseUrl);
      const createAppUrl = setup.createAppUrl;
      // Pass the manifest to be POST'd
      res.render("setup.hbs", { pkg, createAppUrl, manifest });
    });

    route.get("/probot/setup", async (req: Request, res: Response) => {
      const { code } = req.query;
      const response = await setup.createAppFromCode(code);

      // If using glitch, restart the app
      if (process.env.PROJECT_DOMAIN) {
        exec("refresh", (error) => {
          if (error) {
            app.log.error(error);
          }
        });
      } else {
        printRestartMessage(app);
      }

      res.redirect(`${response}/installations/new`);
    });

    route.get("/probot/import", async (_req, res) => {
      const { WEBHOOK_PROXY_URL, GHE_HOST } = process.env;
      const GH_HOST = `https://${GHE_HOST ?? "github.com"}`;
      res.render("import.hbs", { WEBHOOK_PROXY_URL, GH_HOST });
    });

    route.post("/probot/import", bodyParser.json(), async (req, res) => {
      const { appId, pem, webhook_secret } = req.body;
      if (!appId || !pem || !webhook_secret) {
        res.status(400).send("appId and/or pem and/or webhook_secret missing");
        return;
      }
      updateDotenv({
        APP_ID: appId,
        PRIVATE_KEY: `"${pem}"`,
        WEBHOOK_SECRET: webhook_secret,
      });
      res.end();
      printRestartMessage(app);
    });

    route.get("/probot/success", async (req, res) => {
      res.render("success.hbs");
    });

    route.get("/", (req, res, next) => res.redirect("/probot"));
  };

function printWelcomeMessage(
  app: Probot,
  host: string | undefined,
  port: number | undefined
) {
  // use glitch env to get correct domain welcome message
  // https://glitch.com/help/project/
  const domain =
    process.env.PROJECT_DOMAIN ||
    `http://${host ?? "localhost"}:${port || 3000}`;

  [
    ``,
    `Welcome to Probot!`,
    `Probot is in setup mode, webhooks cannot be received and`,
    `custom routes will not work until APP_ID and PRIVATE_KEY`,
    `are configured in .env.`,
    `Please follow the instructions at ${domain} to configure .env.`,
    `Once you are done, restart the server.`,
    ``,
  ].forEach((line) => {
    app.log.info(line);
  });
}

function printRestartMessage(app: Probot) {
  app.log.info("");
  app.log.info("Probot has been set up, please restart the server!");
  app.log.info("");
}

function getBaseUrl(req: Request): string {
  const protocols = req.headers["x-forwarded-proto"] || req.protocol;
  const protocol =
    typeof protocols === "string" ? protocols.split(",")[0] : protocols[0];
  const host = req.headers["x-forwarded-host"] || req.get("host");
  const baseUrl = `${protocol}://${host}`;
  return baseUrl;
}
