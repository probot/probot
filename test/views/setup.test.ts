import { describe, expect, test } from "vitest";

import { setupView } from "../../src/views/setup";

describe("setupView", () => {
  test("providing bare minimum ", () => {
    expect(
      setupView({
        createAppUrl:
          "https://github.com/organizations/probot/settings/apps/new",
        manifest: JSON.stringify({ name: "My App" }),
      }),
    ).toMatchSnapshot();
  });

  test("provide also name", () => {
    expect(
      setupView({
        name: "My App",
        createAppUrl:
          "https://github.com/organizations/probot/settings/apps/new",
        manifest: JSON.stringify({ name: "My App" }),
      }),
    ).toMatchSnapshot();
  });

  test("provide also version", () => {
    expect(
      setupView({
        version: "1.0.0",
        createAppUrl:
          "https://github.com/organizations/probot/settings/apps/new",
        manifest: JSON.stringify({ name: "My App" }),
      }),
    ).toMatchSnapshot();
  });

  test("provide also description", () => {
    expect(
      setupView({
        description: "Awesome App with Probot",
        createAppUrl:
          "https://github.com/organizations/probot/settings/apps/new",
        manifest: JSON.stringify({ name: "My App" }),
      }),
    ).toMatchSnapshot();
  });
});
