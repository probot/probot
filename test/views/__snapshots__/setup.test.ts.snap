// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`setupView > provide also description 1`] = `
"<!DOCTYPE html>
<!DOCTYPE html>
<html lang=\\"en\\" class=\\"height-full\\" data-color-mode=\\"auto\\" data-light-theme=\\"light\\" data-dark-theme=\\"dark\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Setup Your App | built with Probot</title>
    <link rel=\\"icon\\" href=\\"/probot/static/probot-head.png\\">
    <link rel=\\"stylesheet\\" href=\\"/probot/static/primer.css\\">
  </head>
  <body class=\\"height-full bg-gray-light\\">
    <div class=\\"d-flex flex-column flex-justify-center flex-items-center text-center height-full\\">
      <img src=\\"/probot/static/robot.svg\\" alt=\\"Probot Logo\\" width=\\"100\\" class=\\"mb-6\\">
      <div class=\\"box-shadow rounded-2 border p-6 bg-white\\">
        <h1>
          Welcome to your Probot App
          
        </h1>

        <p>Awesome App with Probot</p>

        <div class=\\"text-left mt-6\\">
          <h2 class=\\"alt-h3 mb-2\\">Getting Started</h2>

          <p>To start building a GitHub App, you'll need to register a new app on GitHub.</p>
          <br>

          <form action=\\"https://github.com/organizations/probot/settings/apps/new\\" method=\\"post\\" target=\\"_blank\\" class=\\"d-flex flex-items-center\\">
            <button class=\\"btn btn-outline\\" name=\\"manifest\\" id=\\"manifest\\" value='{\\"name\\":\\"My App\\"}' >Register GitHub App</button>
            <a href=\\"/probot/import\\" class=\\"ml-2\\">or use an existing Github App</a>
          </form>
        </div>
      </div>

      <div class=\\"mt-4\\">
        <h4 class=\\"alt-h4 text-gray-light\\">Need help?</h4>
        <div class=\\"d-flex flex-justify-center mt-2\\">
          <a href=\\"https://probot.github.io/docs/\\" class=\\"btn btn-outline mr-2\\">Documentation</a>
          <a href=\\"https://probot-slackin.herokuapp.com/\\" class=\\"btn btn-outline\\">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`setupView > provide also name 1`] = `
"<!DOCTYPE html>
<!DOCTYPE html>
<html lang=\\"en\\" class=\\"height-full\\" data-color-mode=\\"auto\\" data-light-theme=\\"light\\" data-dark-theme=\\"dark\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Setup My App | built with Probot</title>
    <link rel=\\"icon\\" href=\\"/probot/static/probot-head.png\\">
    <link rel=\\"stylesheet\\" href=\\"/probot/static/primer.css\\">
  </head>
  <body class=\\"height-full bg-gray-light\\">
    <div class=\\"d-flex flex-column flex-justify-center flex-items-center text-center height-full\\">
      <img src=\\"/probot/static/robot.svg\\" alt=\\"Probot Logo\\" width=\\"100\\" class=\\"mb-6\\">
      <div class=\\"box-shadow rounded-2 border p-6 bg-white\\">
        <h1>
          Welcome to My App
          
        </h1>

        <p>This app was built using <a href=\\"https://github.com/probot/probot\\">Probot</a>, a framework for building GitHub Apps.</p>

        <div class=\\"text-left mt-6\\">
          <h2 class=\\"alt-h3 mb-2\\">Getting Started</h2>

          <p>To start building a GitHub App, you'll need to register a new app on GitHub.</p>
          <br>

          <form action=\\"https://github.com/organizations/probot/settings/apps/new\\" method=\\"post\\" target=\\"_blank\\" class=\\"d-flex flex-items-center\\">
            <button class=\\"btn btn-outline\\" name=\\"manifest\\" id=\\"manifest\\" value='{\\"name\\":\\"My App\\"}' >Register GitHub App</button>
            <a href=\\"/probot/import\\" class=\\"ml-2\\">or use an existing Github App</a>
          </form>
        </div>
      </div>

      <div class=\\"mt-4\\">
        <h4 class=\\"alt-h4 text-gray-light\\">Need help?</h4>
        <div class=\\"d-flex flex-justify-center mt-2\\">
          <a href=\\"https://probot.github.io/docs/\\" class=\\"btn btn-outline mr-2\\">Documentation</a>
          <a href=\\"https://probot-slackin.herokuapp.com/\\" class=\\"btn btn-outline\\">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`setupView > provide also version 1`] = `
"<!DOCTYPE html>
<!DOCTYPE html>
<html lang=\\"en\\" class=\\"height-full\\" data-color-mode=\\"auto\\" data-light-theme=\\"light\\" data-dark-theme=\\"dark\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Setup Your App | built with Probot</title>
    <link rel=\\"icon\\" href=\\"/probot/static/probot-head.png\\">
    <link rel=\\"stylesheet\\" href=\\"/probot/static/primer.css\\">
  </head>
  <body class=\\"height-full bg-gray-light\\">
    <div class=\\"d-flex flex-column flex-justify-center flex-items-center text-center height-full\\">
      <img src=\\"/probot/static/robot.svg\\" alt=\\"Probot Logo\\" width=\\"100\\" class=\\"mb-6\\">
      <div class=\\"box-shadow rounded-2 border p-6 bg-white\\">
        <h1>
          Welcome to your Probot App
          <span class=\\"Label Label--outline v-align-middle ml-2 text-gray-light\\">v1.0.0</span>
        </h1>

        <p>This app was built using <a href=\\"https://github.com/probot/probot\\">Probot</a>, a framework for building GitHub Apps.</p>

        <div class=\\"text-left mt-6\\">
          <h2 class=\\"alt-h3 mb-2\\">Getting Started</h2>

          <p>To start building a GitHub App, you'll need to register a new app on GitHub.</p>
          <br>

          <form action=\\"https://github.com/organizations/probot/settings/apps/new\\" method=\\"post\\" target=\\"_blank\\" class=\\"d-flex flex-items-center\\">
            <button class=\\"btn btn-outline\\" name=\\"manifest\\" id=\\"manifest\\" value='{\\"name\\":\\"My App\\"}' >Register GitHub App</button>
            <a href=\\"/probot/import\\" class=\\"ml-2\\">or use an existing Github App</a>
          </form>
        </div>
      </div>

      <div class=\\"mt-4\\">
        <h4 class=\\"alt-h4 text-gray-light\\">Need help?</h4>
        <div class=\\"d-flex flex-justify-center mt-2\\">
          <a href=\\"https://probot.github.io/docs/\\" class=\\"btn btn-outline mr-2\\">Documentation</a>
          <a href=\\"https://probot-slackin.herokuapp.com/\\" class=\\"btn btn-outline\\">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`setupView > providing bare minimum  1`] = `
"<!DOCTYPE html>
<!DOCTYPE html>
<html lang=\\"en\\" class=\\"height-full\\" data-color-mode=\\"auto\\" data-light-theme=\\"light\\" data-dark-theme=\\"dark\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>Setup Your App | built with Probot</title>
    <link rel=\\"icon\\" href=\\"/probot/static/probot-head.png\\">
    <link rel=\\"stylesheet\\" href=\\"/probot/static/primer.css\\">
  </head>
  <body class=\\"height-full bg-gray-light\\">
    <div class=\\"d-flex flex-column flex-justify-center flex-items-center text-center height-full\\">
      <img src=\\"/probot/static/robot.svg\\" alt=\\"Probot Logo\\" width=\\"100\\" class=\\"mb-6\\">
      <div class=\\"box-shadow rounded-2 border p-6 bg-white\\">
        <h1>
          Welcome to your Probot App
          
        </h1>

        <p>This app was built using <a href=\\"https://github.com/probot/probot\\">Probot</a>, a framework for building GitHub Apps.</p>

        <div class=\\"text-left mt-6\\">
          <h2 class=\\"alt-h3 mb-2\\">Getting Started</h2>

          <p>To start building a GitHub App, you'll need to register a new app on GitHub.</p>
          <br>

          <form action=\\"https://github.com/organizations/probot/settings/apps/new\\" method=\\"post\\" target=\\"_blank\\" class=\\"d-flex flex-items-center\\">
            <button class=\\"btn btn-outline\\" name=\\"manifest\\" id=\\"manifest\\" value='{\\"name\\":\\"My App\\"}' >Register GitHub App</button>
            <a href=\\"/probot/import\\" class=\\"ml-2\\">or use an existing Github App</a>
          </form>
        </div>
      </div>

      <div class=\\"mt-4\\">
        <h4 class=\\"alt-h4 text-gray-light\\">Need help?</h4>
        <div class=\\"d-flex flex-justify-center mt-2\\">
          <a href=\\"https://probot.github.io/docs/\\" class=\\"btn btn-outline mr-2\\">Documentation</a>
          <a href=\\"https://probot-slackin.herokuapp.com/\\" class=\\"btn btn-outline\\">Chat on Slack</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;
